pkgs/development/python-modules/nireports/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+  hatch-vcs,
+  hatchling,
+  nipreps-versions,
+  importlib-resources,
+  matplotlib,
+  nibabel,
+  nilearn,
+  nipype,
+  numpy,
+  pandas,
+  pybids,
+  pyyaml,
+  seaborn,
+  svgutils,
+  templateflow,
+  pytestCheckHook,
+  pytest-xdist,
+}:
+
+buildPythonPackage rec {
+  pname = "nireports";
+  version = "23.2.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "nipreps";
+    repo = "nireports";
+    rev = version;
+    hash = "sha256-kr6nWsCA4GcPLEYV/IPm0N3VRz+Mho209/iNVd+59Ik=";
+  };
+
+  build-system = [
+    hatch-vcs
+    hatchling
+    nipreps-versions
+  ];
+
+  dependencies = [
+    matplotlib
+    nibabel
+    nilearn
+    nipype
+    numpy
+    pandas
+    pybids
+    pyyaml
+    seaborn
+    svgutils
+    templateflow
+  ] ++ lib.optionals (pythonOlder "3.12") [
+    importlib-resources
+  ];
+
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeCheckInputs = [
+    matplotlib
+    pytestCheckHook
+    pytest-xdist
+  ];
+
+  pytestFlagsArray = [ "nireports" ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  disabledTests = [
+    # try to download data:
+    "test_cifti_surfaces_plot"
+    "test_mriqc_plot_mosaic"
+    # fails when run under pytest-xdist
+    "test_reportlets.py"
```

Comment:
That won't disable the test, it still being run. I think to disable all tests in a file you have to use `disabledTestPaths` and specify path from repo root.

Another way this problem can be solved, is by forcing pytest-xdist to run things synchronously. Since there aren't that many tests, and they are quick, distribution is not that important. And running more test is a good thing.
But to do that you will have to play with pytest flags to override:
https://github.com/NixOS/nixpkgs/blob/0158c93ad0e497575058ea7553547665c2a27450/pkgs/development/python-modules/pytest-xdist/default.nix#L45

However, specifying whole file in `disabledTestPaths` also is fine.

Reply:
Played around with flags and option. 
Turns out that if you want to run it on 1 worker, you have to **both** disable the hook using `dontUsePytestXdist = true;`, since it will append `--numprocesses=$NIX_BUILD_CORES` overriding the one in `pytestFlagsArray`.
And add a flag `--numprocesses=1` to `pytestFlagsArray`, to override `-n auto` which is specified in their `pyproject.toml` (I think).

If only one them test still run on amount of cores in `/etc/nix/nix.conf`.
When using only 1 core, it took 89 seconds for me, so not too long.